ingestr/src/linkedin_ads/__init__.py

```diff
@@ -0,0 +1,51 @@
+from typing import Iterable
+
+import dlt
+from dlt.common.typing import TDataItem
+from dlt.sources import DltResource
+from pendulum import Date
+
+from .helpers import LinkedInAdsAPI, find_intervals
+
+
+@dlt.source(max_table_nesting=0)
+def linked_in_ads_source(
+    start_date: Date,
+    end_date: Date,
+    access_token: str,
+    account_ids: list[str],
+    dimension: str,
+    metrics: list[str],
+    time_granularity: str,
+) -> DltResource:
+    linkedin_api = LinkedInAdsAPI(
+        access_token=access_token,
+        account_ids=account_ids,
+        dimension=dimension,
+        metrics=metrics,
+        time_granularity=time_granularity,
+    )
+    if time_granularity == "DAILY":
+        primary_key = [dimension, "date"]
```

Comment:
what if `date` is already in `dimension`?

Reply:
It is handled in the source.py file. The dimension will hold only a single value at this time i.e either campaign, creative or account.